test/tests.c

```diff
@@ -293,1421 +264,444 @@ void check_resolve_file(int status, FILE *fd, void *handle)
 
-    storj_download_state_t *state = storj_bridge_resolve_file(env,
-                                                              bucket_id,
-                                                              file_id,
-                                                              download_fp,
-                                                              NULL,
-                                                              check_resolve_file_progress,
-                                                              check_resolve_file_cancel);
+    // get bucket id
+    // NB: bucket id isn't a thing anymore; replacing id with the name.
+    //      Additionally, buckets are always decrypted.
+    status = storj_bridge_get_bucket_id(env, test_bucket_name, NULL, check_get_bucket_id);
+    require_no_last_error_if(status);
+    require_no_last_error_if(uv_run(env->loop, UV_RUN_ONCE));
 
-    if (!state || state->error_status != 0) {
-        return 1;
-    }
+    // upload file
+    reset_test_upload();
+    test_upload(env);
+    require_no_last_error();
+
+    reset_test_upload();
+    test_upload_cancel(env);
+    require_no_last_error();
+
+    // download file
+    reset_test_download();
+    test_download(env);
+    require_no_last_error();
+
+    reset_test_download();
+    test_download_cancel(env);
+    require_no_last_error();
+
+    // list files
+    status = storj_bridge_list_files(env, test_bucket_name,
+                                     &list_options, NULL,
+                                     check_list_files);
+    require_no_last_error_if(status);
+    require_no_last_error_if(uv_run(env->loop, UV_RUN_ONCE));
 
-    // process the loop one at a time so that we can do other things while
-    // the loop is processing, such as cancel the download
-    int count = 0;
-    bool more;
-    int status = 0;
-    do {
-        more = uv_run(env->loop, UV_RUN_ONCE);
-        if (more == false) {
-            more = uv_loop_alive(env->loop);
-            if (uv_run(env->loop, UV_RUN_NOWAIT) != 0) {
-                more = true;
-            }
-        }
+    // get file id
+    // NB: file id isn't a thing anymore; replacing id with the name.
+    status = storj_bridge_get_file_id(env, test_bucket_name, test_upload_file_name,
+                                      NULL, check_get_file_id);
+    require_no_last_error_if(status);
+    require_no_last_error_if(uv_run(env->loop, UV_RUN_ONCE));
 
-        count++;
+    // get file info
+    status = storj_bridge_get_file_info(env, test_bucket_name,test_upload_file_name,
+                                        NULL, check_file_info);
+    require_no_last_error_if(status);
+    require_no_last_error_if(uv_run(env->loop, UV_RUN_ONCE));
 
-        if (count == 100) {
-            status = storj_bridge_resolve_file_cancel(state);
-            assert(status == 0);
-        }
 
-    } while (more == true);
+    // delete a file in a bucket
+    status = storj_bridge_delete_file(env,
```

Comment:
There is a todo... will do, but I would just use the libuplink function instead of libstorj async wrapper. Do you think it makes much of a diff to use `list_objects` vs. `get_object_meta`?

Reply:
https://github.com/storj/libstorj/pull/460/commits/57803b81dafe4f7e989728542c011c60cdf20cfd

Reply:
What you wrote seems fine to me